pkg/kudoctl/util/repo/repo.go

```diff
@@ -0,0 +1,52 @@
+package repo
+
+import (
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+)
+
+// Entry represents a collection of parameters for framework repository
+type Entry struct {
+	Name      string `json:"name"`
+	LocalPath string `json:"localPath"`
+	URL       string `json:"url"`
+}
+
+// Metadata for a Framework. This models the structure of a bundle.yaml file.
+type Metadata struct {
+	// The name of the framework
+	Name string `json:"name,omitempty"`
+	// A SemVer 2 conformant version string of the framework
+	Version string `protobuf:"bytes,4,opt,name=version" json:"version,omitempty"`
+	// The base version of the application enclosed inside of this framework.
+	BaseVersion string `json:"baseVersion,omitempty"`
+	// The KUDO API Version of this framework.
+	APIVersion string `json:"apiVersion,omitempty"`
+	// KubeVersion is a SemVer constraint specifying the version of Kubernetes required.
+	KubeVersion string `json:"kubeVersion,omitempty"`
+	// A SemVer 2 conformant version string of the framework to upgrade from
+	UpgradeVersion string `json:"upgradeVersion,omitempty"`
```

Comment:
what is the idea behind this? 🤔 I say that this can be installed as upgrade only from that specified version?

Reply:
That was the originally idea to start incorporating and using https://github.com/kudobuilder/kudo/blob/a843a200d71b2c343bfc7d2ea7a6dd5aed57f75f/pkg/apis/kudo/v1alpha1/frameworkversion_types.go#L49 as well